File: Backend/AstroGame.Shared/Models/Technologies/FinishedTechnology.cs
using System;

namespace AstroGame.Shared.Models.Technologies
{
    public abstract class FinishedTechnology
    {
        public Guid Id { get; set; }
        public Guid TechnologyId { get; set; }
        public uint Level { get; set; }

        public virtual Technology Technology { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/finished_technologies/finished_technology.dart
import 'package:astrogame_app/communications/converters/finished_technology_converter.dart';
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';

import 'package:json_annotation/json_annotation.dart';

@GuidConverter()
abstract class FinishedTechnology {
  @JsonKey()
  late Guid id;

  @JsonKey()
  late Guid technologyId;

  @JsonKey()
  late Guid playerId;

  @JsonKey()
  late int level;

  FinishedTechnology();

  factory FinishedTechnology.fromJson(Map<String, dynamic> json) =>
      FinishedTechnologyConverter().fromJson(json);
}
